src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,54 @@
 package core.basesyntax.service;
 
+import core.basesyntax.InvalidUserException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int AGE_THRESHOLD = 18;
+    private static final int MIN_NUMBER_OF_CHARS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkForNulls(user);
+        checkIfUserExists(user);
+        checkForValidParameters(user);
```

Comment:
Firstly, you should check if every parameter is valid and then - if the user exists 
```suggestion
        checkForValidParameters(user);
        checkIfUserExists(user);
```